tests/reduction/reduction_get_lambda.h

```diff
@@ -0,0 +1,267 @@
+      }
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer while
+ *         sycl::handler.parallel_for used sycl::nd_range
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
+ *  @tparam AccessorT buffer accessor type
+ *  @param accessor Accessor to the buffer
+ *  @param number_elements Number elements in sycl::span, used only when
+ *         constructing reducer with span
+ *  @retval Lambda with chosen operator
+ */
+template <typename VariableT, typename FunctorT, typename UseCombineFlagT,
+          typename AccessorT>
+auto get_lambda_with_nd_range_for_span(AccessorT accessor,
+                                       size_t number_elements) {
+  if constexpr (UseCombineFlagT::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (int i{}; i < number_elements; i++) {
+        reducer[i].combine(accessor[nd_item.get_global_id()]);
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::multiplies<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (int i{}; i < number_elements; i++) {
+        reducer[i] *= accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT, sycl::plus<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (int i{}; i < number_elements; i++) {
+        reducer[i] += accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT, sycl::bit_or<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (int i{}; i < number_elements; i++) {
+        reducer[i] |= accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_and<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (int i{}; i < number_elements; i++) {
+        reducer[i] &= accessor[nd_item.get_global_id()];
+      }
+    };
+  } else if constexpr (std::is_same<FunctorT,
+                                    sycl::bit_xor<VariableT>>::value) {
+    return [=](sycl::nd_item<1> nd_item, auto &reducer) {
+      for (int i{}; i < number_elements; i++) {
+        reducer[i] ^= accessor[nd_item.get_global_id()];
+      }
+    };
+  }
+}
+
+/** @brief Construct lambda for interacting with reducer in
+ *         sycl::handler.parallel_for with using sycl::span
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam RangeT sycl::range or sycl::nd_range type
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
```

Comment:
```suggestion
 *  @tparam UseCombineFlagT std::integral_constant type that lets switch between
```

Reply:
I've applied this comment in this commit: 39139a01292d8b72557c33b4fa99352e6cf2d58f